custom_components/sat/climate.py

```
@@ -496,24 +496,36 @@ def hvac_action(self):
+        The setpoint is determined primarily by the global heating curve and PID output.
+        If the system is in 'comfort mode' and a focus area is defined with higher priority (based on its error),
+        the setpoint will be overridden by the focus area's heating curve and PID output.
+
+        The setpoint is always constrained to be above the minimum allowable setpoint.
+        """
+        # Default to global heating curve and PID output
+        pid_output = self.pid.output
+        heating_curve = self.heating_curve.value
+
+        # Override with focus area values in comfort mode
+        if self._heating_mode == HEATING_MODE_COMFORT and self.areas.focus is not None and self.areas.focus.error > self.error:
```

Ruff (0.8.2)

`HEATING_MODE_COMFORT` may be undefined, or defined from star imports

(F405)